core/drivers/crypto/caam/include/caam_hal_rng.h

```diff
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2019 NXP
+ *
+ * Brief   CAAM Random Number Generator Hardware Abstration Layer.
+ *         Implementation of primitives to access HW
+ */
+#ifndef __CAAM_HAL_RNG_H__
+#define __CAAM_HAL_RNG_H__
+
+#include <types_ext.h>
+
+/*
+ * Returns if all RNG State Handler already instantiated or not
+ *
+ * @baseaddr  RNG Base Address
+ */
+bool caam_hal_rng_instantiated(vaddr_t baseaddr);
+
+/*
+ * Returns the number of RNG State Handle
+ *
+ * @baseaddr  RNG Base Address
+ */
+uint32_t caam_hal_rng_get_nb_sh(vaddr_t baseaddr);
+
+/*
+ * Returns the RNG Status State Handle
+ *
+ * @baseaddr  RNG Base Address
+ */
+uint32_t caam_hal_rng_get_sh_status(vaddr_t baseaddr);
+
+/*
+ * Returns true if the RNG Key is loaded, false otherwise
+ *
+ * @baseaddr  RNG Base Address
+ */
+bool caam_hal_rng_key_loaded(vaddr_t baseaddr);
+
+/*
+ * Configures the RNG entropy delay
+ *
+ * @baseaddr   RNG Base Address
+ * @inc_delay  Entropy Delay incrementation
+ */
+enum CAAM_Status caam_hal_rng_kick(vaddr_t baseaddr, uint32_t inc_delay);
```

Comment:
Is a fixed width type really needed for `inc_delay`? Which unit used for `inc_delay`, milliseconds, microseconds, nanoseconds, or something else?

Reply:
This is a HW value linked to the CAAM HW configuration. This is not exactly a time in a second unit. This is something like cycle. So here I prefer to keep the uint32_t.